apps/cnquery/cmd/builder/builder.go

```diff
@@ -355,9 +357,19 @@ func dockerProviderCmd(commonCmdFlags commonFlagsFn, preRun commonPreRunFn, runF
 		Use:    "docker ID",
 		Short:  docs.GetShort("docker"),
 		Long:   docs.GetLong("docker"),
-		Args:   cobra.ExactArgs(1),
+		Args:   cobra.MaximumNArgs(1),
 		PreRun: preRun,
 		Run: func(cmd *cobra.Command, args []string) {
+			discover, err := cmd.Flags().GetString("discover")
+			if err != nil {
+				log.Error().Err(err).Msg("failed to retrieve discover flag")
+				return
+			}
+			if len(args) == 0 && !strings.Contains(discover, docker_engine.DiscoveryContainerRunning) {
```

Comment:
what happens if image discovery is active?

Reply:
I allowed all flag but `auto` since that is the only one that does nothing (and the empty flag)